File: competition/13-regex/regex.rb
def line(i)
  a = (i * 7) % 100000
  b = (i * 13) % 100000
  c = (i * 31) % 100000
  "user=#{a} noise word here id=#{b} and ok=#{c}"
end

text = (0...12000).map { |i| line(i) }.join("\n")
re = /([A-Za-z]+)=([0-9]+)/
count = 0
total = 0
text.scan(re) do
  m = $~
  count += 1
  total = (total + m[0].length) % 1000000007
end
puts count
puts total

File: competition/13-regex/regex.js
function line(i) {
  const a = (i * 7) % 100000;
  const b = (i * 13) % 100000;
  const c = (i * 31) % 100000;
  return `user=${a} noise word here id=${b} and ok=${c}`;
}

const parts = [];
for (let i = 0; i < 12000; i++) parts.push(line(i));
const text = parts.join("\n");

const re = /([A-Za-z]+)=([0-9]+)/g;
let count = 0;
let total = 0;
let m;
while ((m = re.exec(text)) !== null) {
  count++;
  total = (total + m[0].length) % 1000000007;
}
console.log(count);
console.log(total);
